pkg/rancher-desktop/backend/kube/lima.ts

```diff
@@ -125,9 +126,11 @@ export default class LimaKubernetesBackend extends events.EventEmitter implement
    * Install the Kubernetes files.
    */
   async install(config: BackendSettings, desiredVersion: semver.SemVer, allowSudo: boolean) {
+    const configWithK8sVersion = config.kubernetes.version ? config : merge({}, config, { kubernetes: { version: desiredVersion.version } });
```

Comment:
We set `activeVersion` (line 133/136) directly to `desiredVersion`, so that should be the version we use (ignoring the config).

Of course, that means we're passing the wrong version to `this.writeServiceScript`. It might be better for the caller to pass in the fixed-up config here instead?

(I also don't understand how this fixes the issue, since your original changes disappeared…)

Reply:
The issue is that there's a surfaced error message coming from `BackendHelper.requiresCRIDockerd(cfg.containerEngine.name, cfg.kubernetes.version)` which is unhappy because `cfg.kubernetes.version` is the empty string at startup.

This field probably should be `desiredVersion` all along, but someone tried an optimization to use the version field off of `config` and save passing an argument. I'm just guessing here, but maybe the proper fix is to pass `desiredVersion` all that way and  wait for `config.kubernetes.version` to be updated normally via message passing?

Reply:
Yeah, it sounds like passing `desiredVersion` to `requiresCRIDockerd` is the correct fix.

This fix is incorrect because if we had a non-empty (but unavailable) k8s version in the configs, we'll pass the wrong value in and `requiresCRIDockerd` will make the wrong decision.  For example, if the user is offline, has a _different_ version of k3s cached, but the version in the config is not cached.